pkg/helpers/helpers.go

```diff
@@ -246,3 +246,57 @@ func FindTaintByKey(managedCluster *clusterv1.ManagedCluster, key string) *clust
 	}
 	return nil
 }
+
+func IsTaintEqual(taint1, taint2 clusterv1.Taint) bool {
+	// Ignore the comparison of time
+	return taint1.Key == taint2.Key && taint1.Value == taint2.Value && taint1.Effect == taint2.Effect
+}
+
+func IsTaintsEqual(taints1 []clusterv1.Taint, taints2 []clusterv1.Taint) bool {
+	if len(taints1) != len(taints2) {
+		return false
+	}
+	if (taints1 == nil) != (taints2 == nil) {
+		return false
+	}
+	if len(taints2) == 0 && len(taints1) == 0 {
+		return true
+	}
+	for i, v := range taints1 {
+		if !IsTaintEqual(taints2[i], v) {
+			return false
+		}
+	}
+
+	return true
+}
+
+// AddTaints add taints to the specified slice, if it did not already exist.
+// Return a boolean indicating whether the slice has been updated.
+func AddTaints(taints *[]clusterv1.Taint, taint clusterv1.Taint) bool {
+	for _, v := range *taints {
+		if IsTaintEqual(v, taint) {
+			return false
+		}
+	}
+	*taints = append(*taints, taint)
+	return true
+}
+
+func RemoveTaints(taints *[]clusterv1.Taint, targets ...clusterv1.Taint) (updated bool) {
+	var newTaints []clusterv1.Taint
+	for _, v1 := range *taints {
+		found := false
+		for _, v2 := range targets {
```

Comment:
you could make it another func in this for as
```
FindTaints(taints []clusterv1.Taint, taint clusterv1.Taint) *clusterv1.Taint
```
This could be used in both AddTaint and RemoveTaint